cluster/provider.go

```diff
@@ -8,4 +13,6 @@ type Provider interface {
 	GetAdvertiseAddress() string
 	GetRemoteAddress() string
 	ListenClusterEvents() <-chan struct{}
+	AllocateEndpoint(string, string, *network.EndpointSettings) (string, types.NetworkCreateRequest, error)
```

Comment:
I dont think exposing libnetwork with engine-api is a good idea. Especially to address a single feature when this can be avoided.